include/odp/api/spec/classification.h

```diff
@@ -644,6 +682,29 @@ void odp_cls_pmr_param_init(odp_pmr_param_t *param);
 odp_pmr_t odp_cls_pmr_create(const odp_pmr_param_t *terms, int num_terms,
 			     odp_cos_t src_cos, odp_cos_t dst_cos);
 
+/**
+ * Create a packet matching rule with options
+ *
+ * Similar to odp_cls_pmr_create() function with additional PMR creation
+ * options specified through odp_pmr_create_opt_t.
+ *
+ * Use odp_cls_pmr_create_opt_init() to initialize options into their default
+ * values.
+ *
+ * @param opt	       points to PMR create options
+ * @param src_cos      source CoS handle
+ * @param dst_cos      destination CoS handle
+ *
+ * @return Handle to the Packet Match Rule.
+ * @retval ODP_PMR_INVALID on failure
+ *
+ * @note Application can mix the usage of odp_cls_pmr_create() and
```

Comment:
I would actually add this in the documentation of odp_cls_pmr_create():

"Matching PMR rule created through this function sets the CLS mark metadata of the packet to zero.

Rules created through this function are equivalent to rules created through odp_cls_pmr_create() with the same PMR terms and with the additional option fields set to their default values."

After that, I am not sure if this note is needed anymore. If it is, the grammar needs to be fixed. I would simply say: "Rules can be created through both odp_cls_pmr_create() odp_cls_pmr_create_opt() in the same application."

Reply:
makes sense. Update odp_cls_pmr_create()'s note to.
+ *
+ * @note Matching PMR rule created through this function sets the CLS mark metadata
+ * of the packet to zero.
+ *
+ * @note Rules created through this function are equivalent to rules created through
+ * odp_cls_pmr_create() with the same PMR terms and with the additional option fields
+ * set to their default values.